packages/fluentui/react/src/components/Flex/Flex.tsx

```diff
@@ -38,63 +41,93 @@ export interface FlexProps extends UIComponentProps, ChildrenComponentProps {
   fill?: boolean;
 }
 
-class Flex extends UIComponent<WithAsProp<FlexProps>> {
-  static Item = FlexItem;
+export type FlexStylesProps = Pick<
+  FlexProps,
+  'column' | 'debug' | 'fill' | 'gap' | 'hAlign' | 'inline' | 'padding' | 'space' | 'vAlign' | 'wrap'
+>;
+
+const Flex: React.FC<WithAsProp<FlexProps>> & { className: string; handledProps: (keyof FlexProps)[]; Item: typeof FlexItem } = props => {
+  const context: ProviderContextPrepared = React.useContext(ThemeContext);
+  const { setStart, setEnd } = useTelemetry(Flex.displayName, context.telemetry);
+  setStart();
+
+  const { children, className, column, debug, design, fill, gap, hAlign, inline, padding, space, styles, variables, vAlign, wrap } = props;
+
+  const { classes } = useStyles<FlexStylesProps>(Flex.displayName, {
+    className: Flex.className,
+    mapPropsToStyles: () => ({
+      column,
+      debug,
+      fill,
+      gap,
+      hAlign,
+      inline,
+      padding,
+      space,
+      vAlign,
+      wrap
+    }),
+    mapPropsToInlineStyles: () => ({
+      className,
+      design,
+      styles,
+      variables
+    }),
+    rtl: context.rtl
+  });
 
-  static displayName = 'Flex';
-  static className = 'ui-flex';
+  const ElementType = getElementType(props);
+  const unhandledProps = getUnhandledProps(Flex.handledProps, props);
 
-  static defaultProps = {
-    as: 'div'
-  };
+  const content = React.Children.map(children, child => {
```

Comment:
Let's refactor this with adding `FlexContextProvider`

Reply:
Agree, can we do this thing separately? I want to make a small showcase from it